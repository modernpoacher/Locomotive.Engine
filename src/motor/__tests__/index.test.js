import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Motor from 'shinkansen-motor/motor'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('react-jsonschema-form')

describe('shinkansen-motor/motor', () => {
  describe('Always', () => {
    it('renders', () => {
      const component = (
        <Motor definition={{}} />
      )

      expect(renderer.create(component).toJSON())
        .toMatchSnapshot()
    })
  })

  describe('`definition` has `schema` object', () => {
    it('renders', () => {
      const component = (
        <Motor definition={{ schema: {} }} />
      )

      expect(renderer.create(component).toJSON())
        .toMatchSnapshot()
    })
  })

  describe('`shouldComponentUpdate()`', () => {
    const schema = {}

    let instance

    beforeEach(() => {
      const component = (
        <Motor definition={{ schema }} />
      )

      const wrapper = shallow(component)

      instance = wrapper.instance()
    })

    describe('`props` has changed', () => {
      it('returns true', () => {
        expect(instance.shouldComponentUpdate({ definition: { schema: {} } }))
          .toBe(true)
      })
    })

    describe('`props` has not changed', () => {
      it('returns false', () => {
        expect(instance.shouldComponentUpdate({ definition: { schema } }))
          .toBe(false)
      })
    })
  })
})
