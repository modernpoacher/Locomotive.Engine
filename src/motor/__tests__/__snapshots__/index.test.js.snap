// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shinkansen-motor/motor \`definition\` has \`schema\` object renders 1`] = `
<div
  className="shinkansen-motor"
/>
`;

exports[`shinkansen-motor/motor Always renders 1`] = `null`;
